code/_onclick/hud/interactive_emotes.dm

```diff
@@ -0,0 +1,279 @@
+				time = 0.5 SECONDS, easing = BACK_EASING, flags = ANIMATION_PARALLEL)
+	animate(pixel_x = initial(owner.pixel_x), pixel_y = initial(owner.pixel_y), time = 0.1 SECONDS)
+
+	if(isxeno(initiator))
+		animate(initiator, pixel_x = initiator.pixel_x + x_distance * 8, pixel_y = initiator.pixel_y + y_distance * 8, dir = owner.dir,
+				time = 0.5 SECONDS, easing = BACK_EASING, flags = ANIMATION_PARALLEL)
+		initiator.face_atom(owner)
+	else
+		animate(initiator, pixel_x = initiator.pixel_x + x_distance * 8, pixel_y = initiator.pixel_y + y_distance * 8,
+				time = 0.5 SECONDS, easing = BACK_EASING, flags = ANIMATION_PARALLEL)
+	animate(pixel_x = initial(initiator.pixel_x), pixel_y = initial(initiator.pixel_y), time = 0.1 SECONDS)
+
+//We support interspecies bumping of fists and tails!
+/atom/movable/screen/interaction/fist_bump/success_message()
+	var/owner_xeno = isxeno(owner)
+	var/initiator_xeno = isxeno(initiator)
+	if(owner_xeno && initiator_xeno)
+		return "[owner] and [initiator] slap tails together!"
+	if(owner_xeno)
+		return "[owner] slaps [initiator]'s fist!"
+	if(initiator_xeno)
+		return "[owner] fist bumps [initiator]'s tail!"
+	return "[owner] fist bumps [initiator]!"
+
+/atom/movable/screen/interaction/fist_bump/failure_message()
+	return "[owner] left [initiator] hanging. Not cool!"
+
+/atom/movable/screen/interaction/headbutt
+	name = "head bump"
+	desc = "Touch skulls."
+	interaction_sound = 'sound/weapons/throwtap.ogg'
+
+/atom/movable/screen/interaction/headbutt/interaction_animation()
+	owner.face_atom(initiator)
+	initiator.face_atom(owner)
+
+	var/x_distance = owner.x - initiator.x
+	var/y_distance = owner.y - initiator.y
+
+	var/matrix/owner_matrix = owner.transform
+	var/matrix/initiator_matrix = initiator.transform
+	var/rotation_angle
+	//This was so much pain, maintainers forgive me
+	if(owner.dir & (EAST | WEST))
+		if(isxenorunner(owner))	//Rounies get special upwards headbutts
+			rotation_angle = owner.dir & EAST ? -15 : 15
+		else
+			rotation_angle = owner.dir & EAST ? 15 : -15
+
+		//The animation if the mobs face east/west is to rotate their heads together
+		animate(owner, pixel_x = owner.pixel_x - x_distance * 8, pixel_y = owner.pixel_y - y_distance * 8,
+				transform = owner_matrix.Turn(rotation_angle), time = 0.5 SECONDS, easing = BACK_EASING, flags = ANIMATION_PARALLEL)
+		animate(pixel_x = initial(owner.pixel_x), pixel_y = initial(owner.pixel_y),
+				transform = owner_matrix.Turn(-rotation_angle), time = 0.1 SECONDS)
+	else
+		//If facing north or south, basically the same animation as the high five but move even closer
+		animate(owner, pixel_x = owner.pixel_x - x_distance * 8, pixel_y = owner.pixel_y - y_distance * 16,
+				time = 0.5 SECONDS, easing = BACK_EASING, flags = ANIMATION_PARALLEL)
+		animate(pixel_x = initial(owner.pixel_x), pixel_y = initial(owner.pixel_y), time = 0.1 SECONDS)
+
+	if(initiator.dir & (EAST | WEST))
```

Comment:
dir isnt a bitfield? You can't do this

Reply:
I was just copying how every other place in the code does it. For example: barricades. And it works so... 🤷

Reply:
Yeah nevermind I'm dumb